Mathlib/Data/Matroid/Rank/Finite.lean

```diff
@@ -0,0 +1,190 @@
+/-
+Copyright (c) 2025 Peter Nelson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Peter Nelson
+-/
+import Mathlib.Data.Matroid.Closure
+
+/-!
+# Finite-rank sets
+
+For `M : Matroid α` and `X : Set α`,
+`Matroid.RkFin M X`  means that every basis of `X` in `M` is finite,
+or equivalently that the restriction to `X` is `Matroid.RankFinite`.
+`RkFin` sets in a matroid are the largest class of sets for which one can do nontrivial
+integer arithmetic involving the rank function.
+
+# Implementation Details
+
+Unlike most set predicates on matroids, a set `X` with `M.RkFin X` need not satisfy `X ⊆ M.E`,
+so may contain junk elements. This seems to be what makes the definition easiest to use.
+-/
+
+variable {α : Type*} {M : Matroid α} {X Y I : Set α} {e : α}
+
+open Set
+
+namespace Matroid
+
+/-- a `RkFin` set in `M` is one whose bases are all finite. -/
+def RkFin (M : Matroid α) (X : Set α) : Prop := (M ↾ X).RankFinite
```

Comment:
Note that this is another exception to the `Is`-naming convention (eg discussed here https://leanprover.zulipchat.com/#narrow/channel/287929-mathlib4/topic/Naming.20conventions.20for.20.60Prop.60-valued.20defs.20and.20classes). I find some of the matroid lemmas confusing to read because of this - particularly the ones involving `Basis` and `Circuit`: `Basis` in other parts of mathlib is the data of a basis, whereas here it's the proposition that a set is a basis! Please consider renaming the matroid ones sooner rather than later, to avoid it being too time-consuming to adjust to.

Reply:
I do take this point. Do you think it is ok to draw the line at adjectives, though? `IsIndep` makes my fingers unhappy.

Reply:
Yeah, I think that's fine